### PR TITLE
Is valid end output error

### LINKED ISSUE #132: Throw ValidEndDate error from parser.
Throw error from parser level

### PR DESCRIPTION
**Pull request by nowknowing**
output error to user

merge test branch first so that difference is only in 1 file

### CODE REVIEW

File: src/test/java/seedu/address/model/session/RecurringSessionTest.java
Comment (enhao25) [inline src/test/java/seedu/address/model/session/RecurringSessionTest.java]: Hi, was thinking that maybe a variable name such as consistentDate1 might be more meaningful here.

File: src/test/java/seedu/address/model/session/RecurringSessionTest.java
Comment (enhao25) [inline src/test/java/seedu/address/model/session/RecurringSessionTest.java]: Hi, maybe using something like "getIntervalTest" might make it more explicate that its a method for testing.

File: src/test/java/seedu/address/testutil/RecurringSessionBuilder.java
Comment (enhao25) [inline src/test/java/seedu/address/testutil/RecurringSessionBuilder.java]: Hi, I think you have missed out on writing the description for the return method here.

File: src/test/java/seedu/address/testutil/TypicalStudents.java
Comment (enhao25) [inline src/test/java/seedu/address/testutil/TypicalStudents.java:49]: One of my PR changes the method from .addSession to .addSessions instead, we might get some conflict because of that.

File: src/test/java/seedu/address/model/session/RecurringSessionTest.java
Comment (enhao25) [inline src/test/java/seedu/address/model/session/RecurringSessionTest.java]: Similar to the review above.

File: src/main/java/seedu/address/model/session/RecurringSession.java
Comment (enhao25) [inline src/main/java/seedu/address/model/session/RecurringSession.java]: Hi, actually I am abit confused between this isConsistent method compared to the isValidEnd method based on the description. One mention that it returns true if it makes a valid lastSessionDate while the other returns true if it is consistent with the lastSessionDate. The 2 descriptions seems very similar in my opinion.

**Review by enhao25 — COMMENTED** [review]
LGTM! PRed the application and tried it out. Seems good. Just had some questions and comments for certain parts.

File: src/test/java/seedu/address/testutil/TypicalStudents.java
Comment (nowknowing) [reply src/test/java/seedu/address/testutil/TypicalStudents.java:49]: ok will wait for that to merge first then

File: src/main/java/seedu/address/model/session/RecurringSession.java
Comment (nowknowing) [reply src/main/java/seedu/address/model/session/RecurringSession.java]: yeah the only difference is if it checks the date vs both date and time

File: src/test/java/seedu/address/model/session/RecurringSessionTest.java
Comment (enhao25) [inline src/test/java/seedu/address/model/session/RecurringSessionTest.java]: Hi maybe this can be session instead of s. Just worried that using s instead would violate the java coding standard.

File: src/main/java/seedu/address/model/session/RecurringSession.java
Comment (enhao25) [reply src/main/java/seedu/address/model/session/RecurringSession.java]: Maybe it would be better to include this difference in the description as well. Because I think if u didn't tell me this, I would not have known from the description alone.

**Review by enhao25 — APPROVED** [review]
LGTM. Just 1 comment about using "s" as a variable name. Also on the previous comment to explain in more detail the isConsistent and isValidEnd method in RecurringSession.java. After updating that, feel free to merge it in.

**Pull request merged**
